### PR TITLE
Add Vasil hardfork to cardano-api and cardano-cli

### PR DESCRIPTION
**Pull request by Jimbo4350**
This PR only introduces the era. It does not begin to integrate the new functionality exposed by `cardano-ledger`.

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Tx.hs [hunk 1]
```diff
@@ -373,6 +393,7 @@ instance IsCardanoEra era => SerialiseAsCBOR (KeyWitness era) where
         AllegraEra -> decodeShelleyBasedWitness ShelleyBasedEraAllegra bs
         MaryEra    -> decodeShelleyBasedWitness ShelleyBasedEraMary    bs
         AlonzoEra  -> decodeShelleyBasedWitness ShelleyBasedEraAlonzo  bs
+        BabbageEra -> decodeShelleyBasedWitness ShelleyBasedEraBabbage  bs
```
Comment (Unisay) [inline cardano-api/src/Cardano/Api/Tx.hs:396]:
```suggestion
        BabbageEra -> decodeShelleyBasedWitness ShelleyBasedEraBabbage bs
```

File: cardano-api/src/Cardano/Api/TxBody.hs [hunk 2]
```diff
@@ -518,6 +526,12 @@ instance IsCardanoEra era => ToJSON (TxOut ctx era) where
            , "datumhash" .= h
            , "datum"     .= scriptDataToJson ScriptDataJsonDetailedSchema d
            ]
+  toJSON (TxOut addr val (TxOutInlineDatum _ d)) =
+    object [ "address"   .= addr
+           , "value"     .= val
+           , "datumhash" .= hashScriptData d
```
[the remark this answers is not in the document]
Comment (Jimbo4350) [reply cardano-api/src/Cardano/Api/TxBody.hs:532]: Ah that's a good point. What about "inlinedatumhash" and "inlinedatum"?

[the remark this answers is not in the document]
Comment (JaredCorduan) [reply cardano-api/src/Cardano/Api/TxBody.hs:532]: sounds great!

**Review by MarcFontaine — APPROVED** [review]
LGTM

**Comment by JaredCorduan** [comment]
one note  about names: it is the Vasil hardfork, but the Babbage era :grimacing:

**Pull request merged**